scripts/devops_tasks/tox_harness.py

```diff
@@ -119,17 +119,18 @@ def replace_dev_reqs(file, pkg_root):
         for line in f:
             args = [part.strip() for part in line.split() if part and not part.strip() == "-e"]
             amended_line = " ".join(args)
+            extras = ""
 
             if amended_line.endswith("]"):
-                trim_amount = amended_line[::-1].index("[") + 1
-                amended_line = amended_line[0 : (len(amended_line) - trim_amount)]
+                amended_line, extras = amended_line.rsplit("[", maxsplit=1)
+                if extras:
+                    extras = f"[{extras}"
 
-            adjusted_req_lines.append(amended_line)
+            adjusted_req_lines.append(f"{build_whl_for_req(amended_line, pkg_root)}{extras}")
 
     req_file_name = os.path.basename(file)
     logging.info("Old {0}:{1}".format(req_file_name, adjusted_req_lines))
 
-    adjusted_req_lines = list(map(lambda x: build_whl_for_req(x, pkg_root), adjusted_req_lines))
```

Comment:
While I'd factor it differently now, what I was thinking when originally writing this code was to give a place where we print the lines before and after we adjust them. In this case, if you look at the log of `old` and `new` they will be the same if you pull this into the loop.  I'd like to keep that around, I'm betting this is how you located that those dev reqs were being replaced.

Reply:
My bad, I thought I'd kept that behavior as it was!

---

Tentatively resolved by: https://github.com/Azure/azure-sdk-for-python/pull/32864/commits/eae1d0cb58dc6097d297c8b61227f32ba883cf21

The behavior is _slightly_ different. The "old" log message wouldn't show any `-e` that were in the original requirement. But the above commit preserves them.

Reply:
Honestly that's better, so I won't complain. 👍